reaktive/src/jsCommonTest/kotlin/com/badoo/reaktive/scheduler/MainSchedulerTest.kt

```diff
@@ -9,4 +20,62 @@ class MainSchedulerTest {
         val executor = scheduler.newExecutor()
         executor.submit {}
     }
+
+    @Test
+    fun delayed_task_executed_with_delay(): AsyncTestResult {
+        val scheduler = MainScheduler()
+        val executor = scheduler.newExecutor()
+        var executed = false
+        executor.submit(200.milliseconds) { executed = true }
+
+        return observableOf(
+            100.milliseconds, 300.milliseconds
+        )
+            .flatMapSingle { timeout ->
+                singleTimer(timeout, scheduler = scheduler)
+                    .map { executed }
+            }
+            .toList()
+            .testAwait { (before, after) ->
+                assertFalse(before)
+                assertTrue(after)
+            }
+    }
+
+    @Test
+    fun interval_task_executed_at_interval(): AsyncTestResult {
+        val scheduler = MainScheduler()
+        val executor = scheduler.newExecutor()
+        var counter = 0L
+        val items = mutableListOf<Long>()
+
+        val delayBetweenTasks = 50.milliseconds
+
+        executor.submit(period = delayBetweenTasks) {
+            items.add(counter++)
+        }
+
+        val amountToTest = 10
+        val testTickShift = 25L
+
+        val testTicks = 1..amountToTest
+        val expectedItems = testTicks.map { tick ->
+            (0 until tick).map { it.toLong() }.toList()
+        }.toList()
+
+        return observableOf(
+            *testTicks
+                .map { it * 50L + testTickShift }
```

Comment:
It takes a lot of time to understand the test. I suggest to just specify all durations as a simple list, similarly to the test above. E.g. `observableOf(25.milliseconds, 75.milliseconds, ...)`. And I think 5 ticks should be enough.

Reply:
done